module/zfs/dmu.c

```diff
@@ -725,12 +725,8 @@ get_next_chunk(dnode_t *dn, uint64_t *start, uint64_t minimum, uint64_t *l1blks)
 	ASSERT3U(minimum, <=, *start);
 
 	if (*start - minimum <= iblkrange * maxblks) {
+		*l1blks = DIV_ROUND_UP((*start - minimum), iblkrange);
```

Comment:
I think it's also worth retaining the comment explaining that we are assuming that all the L1 blocks are full.  This is in contrast to the general case which actually looks at the L1 BP's to see how many exist.

I think it might be more clear to do something like:
```
uint64_t total_l1blks = DIV_ROUND_UP((*start - minimum), iblkrange);
if (total_l1blks <= maxblks) {
  *l1blks = total_l1blks;
  *start = minimum;
  return (0);
}
```

Although I think if we want to be totally accurate, something like (not working code):
```
uint64_t total_l1blks = (ROUND_DOWN(*start, iblkrange) - ROUND_UP(minimum, iblkrange)) / iblkrange;
```
or a simpler version, the +2 is for the first and last (potentially) partial L1's:
```
uint64_t total_l1blks = (*start - minimum) / iblkrange + 2;
```

Reply:
@ahrens I think you have the rounding mixed up in your second snippet, correct?

Reply:
Yes, sorry.  Keep forgetting that the range is [minimum, *start].  Should be `(ROUND_UP(*start, iblkrange) - ROUND_DOWN(minimum, iblkrange))`